core/src/main/kotlin/transformers/documentation/DefaultDocumentationNodeMerger.kt

```diff
@@ -64,17 +64,34 @@ fun Property.mergeWith(other: Property) = Property(
     (actual + other.actual).merge()
 )
 
-fun Class.mergeWith(other: Class) = Class(
-    dri,
-    name,
-    kind,
-    merge(constructors + other.constructors, Function::mergeWith),
-    merge(functions + other.functions, Function::mergeWith),
-    merge(properties + other.properties, Property::mergeWith),
-    merge(classes + other.classes, Class::mergeWith),
-    expected?.mergeWith(other.expected),
-    (actual + other.actual).mergeClassPlatformInfo()
-)
+fun ClassLike.mergeWith(other: ClassLike): ClassLike = when {
+        this is Class -> {
+            val constructors = this.constructors + if (other is Class) other.constructors else emptyList()
```

Comment:
Check for `other` shouldn't be done here

Reply:
Changed.